src/pages/projectSettings/tabs/NotificationsTab/utils.ts

```diff
@@ -0,0 +1,153 @@
+    case NotificationMethods.EMAIL:
+      return notification.emailInput;
+    case NotificationMethods.WEBHOOK:
+      return notification.webhookInput.urlInput;
+    case NotificationMethods.JIRA_ISSUE:
+      return notification.jiraIssueInput.projectInput;
+    default:
+      return "";
+  }
+};
+
+// Converts the form regexSelector into the proper format for GQL payload.
+// We need to check if the trigger has regex selectors because it's possible for data
+// from other dependencies to persist.
+const regexFormToGql = (
+  hasRegexSelectors: boolean,
+  regexForm: FormRegexSelector[]
+) =>
+  hasRegexSelectors
+    ? regexForm?.map((r) => ({
+        type: r.regexSelect,
+        data: r.regexInput,
+      }))
+    : [];
+
+const webhookFormToGql = (webhookInput: Notification["webhookInput"]) => {
+  if (!webhookInput) {
+    return null;
+  }
+  return {
+    url: webhookInput.urlInput,
+    secret: webhookInput.secretInput,
+    headers: webhookInput.httpHeaders?.map(({ keyInput, valueInput }) => ({
+      key: keyInput,
+      value: valueInput,
+    })),
+  };
+};
+const jiraFormToGql = (jiraInput: Notification["jiraIssueInput"]) => {
+  if (!jiraInput) {
+    return null;
+  }
+  return {
+    project: jiraInput.projectInput,
+    issueType: jiraInput.issueInput,
+  };
+};
+
+// Converts the form extraFields into the proper format for GQL payload.
+// We need to check what extraFields exist for a particular trigger because it's possible
+// for data from other dependencies to persist.
+const extraFieldsFormToGql = (
+  extraFieldsToInclude: ExtraField[],
+  extraFieldsForm: FormExtraFields
+) => {
+  // If there are no extra fields for this trigger, just return.
+  if (!extraFieldsToInclude) {
+    return {};
+  }
+  const extraFields = {};
+  extraFieldsToInclude.forEach((e) => {
+    extraFields[e.key] = extraFieldsForm[e.key].toString();
+  });
+  return extraFields;
+};
+
+export const getGqlPayload = (
+  subscription: Unpacked<FormState["subscriptions"]>
+): SubscriptionInput => {
+  const { subscriptionData } = subscription;
+  const event = projectTriggers[subscriptionData.event.eventSelect];
+  const { resourceType = "", trigger, extraFields, regexSelectors } =
+    event || {};
+
+  const triggerData = extraFieldsFormToGql(
+    extraFields,
+    subscriptionData.event.extraFields
+  );
+  const selectors = Object.entries(triggerData).map(([key, value]) => ({
+    type: key,
+    data: value.toString(),
+  }));
+  const regexData = regexFormToGql(
+    !!regexSelectors,
+    subscriptionData.event.regexSelector
+  );
+
+  const method = subscriptionData.notification.notificationSelect;
+  const subscriber = getTargetForMethod(method, subscriptionData?.notification);
+  return {
+    trigger,
+    resource_type: resourceType,
+    selectors: [
+      { type: "project", data: resourceType.toLowerCase() },
+      ...selectors,
+    ],
+    trigger_data: triggerData,
+    regex_selectors: regexData,
```

Comment:
I get this error
![image](https://user-images.githubusercontent.com/13104717/178579333-6d33f4a6-f08f-4373-9c65-59634e0bc385.png)

 when trying to save this
![image](https://user-images.githubusercontent.com/13104717/178579249-43eef0d1-3373-4e4b-b7ad-d15f271e7037.png)

despite there not being any regex selectors

Reply:
Fixed